standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java

```diff
@@ -936,6 +921,9 @@ private TTransport createAuthBinaryTransport(URI store, TTransport underlyingTra
       } catch (IOException | TTransportException sasle) {
         // IOException covers SaslException
         LOG.error("Could not create client transport", sasle);
+        if (sasle instanceof TTransportException) {
```

Comment:
why is this hack? could we handle IOException & TTransportException separately?

Reply:
Address.